inspirehep/dojson/jobs/fields/bd1xx.py

```diff
@@ -29,36 +29,41 @@
 from dojson import utils
 
 from ..model import jobs
-from ...utils import classify_rank, get_record_ref
+from ...utils import classify_rank, force_single_element, get_record_ref
 
 
 @jobs.over('date_closed', '^046..')
 def date_closed(self, key, value):
     """Date the job was closed."""
-    value = utils.force_list(value)
-    closed_date = None
-    deadline_date = None
-    for val in value:
-        if val.get('i'):
-            deadline_date = val.get('i')
-        if val.get('l'):
-            if "@" in val.get('l'):
-                # Its an email
-                if "reference_email" in self:
-                    self["reference_email"].append(val.get('l'))
-                else:
-                    self["reference_email"] = [val.get('l')]
-            elif "www" in val.get('l') or "http" in val.get('l'):
-                # Its a URL
-                if "urls" in self:
-                    self["urls"].append(val.get('l'))
-                else:
-                    self["urls"] = [val.get('l')]
-            else:
-                closed_date = val.get('l')
+    def _contains_email(val):
+        return '@' in val
+
+    def _contains_url(val):
+        return 'www' in val or 'http' in val
+
+    el = force_single_element(value)
+
+    deadline_date = el.get('i')
```

Comment:
I would also filter a single element out of `el['i']` and `el['1']`. I didn't check against repeatable fields but this should help a lot